File: tests/e2e/9_cgi/requirements/webserv/volume/www/cgi/hello.php
#!/usr/bin/php-cgi
<?php
header("Content-Type: text/html");
?>
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>PHP CGI</title>
    <style>
        body {
            margin: 0;
            background: #000000;
            color: #ffffff;
            font-family: Helvetica, Arial, sans-serif;
            padding: 2rem;
        }
        .container {
            max-width: 800px;
            margin: 0 auto;
        }
        h1 {
            font-size: 3rem;
            margin: 0 0 2rem 0;
            text-align: center;
        }
        .info {
            margin-bottom: 2rem;
            text-align: center;
        }
        table {
            width: 100%;
            border-collapse: collapse;
        }
        th, td {
            padding: 0.5rem;
            text-align: left;
            border-bottom: 1px solid rgba(255, 255, 255, 0.2);
        }
        th {
            opacity: 0.7;
        }
        a {
            color: #ffffff;
            text-decoration: none;
        }
        a:hover {
            text-decoration: underline;
        }
        .back-link {
            display: block;
            margin-top: 2rem;
            text-align: center;
        }
    </style>
</head>
<body>
    <div class="container">
        <h1>PHP CGI</h1>
        
        <div class="info">
            <p>Time: <?php echo date('Y-m-d H:i:s'); ?></p>
            <p>Version: <?php echo phpversion(); ?></p>
        </div>
        
        <table>
            <tr><th>Variable</th><th>Value</th></tr>
            <?php foreach ($_SERVER as $key => $value): ?>
                <tr>
                    <td><?php echo htmlspecialchars($key); ?></td>
                    <td><?php echo htmlspecialchars($value); ?></td>
                </tr>
            <?php endforeach; ?>
        </table>
        
        <div class="back-link">
            <a href="/">Back</a>
        </div>
    </div>
</body>
</html>

File: tests/e2e/9_cgi/requirements/webserv/volume/www/cgi/hello.cpp
#!/ usr / bin / env python3
import sys import datetime

    sys.stdout.write("Content-Type: text/html\r\n") sys.stdout.write("\r\n")

                                                                         current_time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")

                                                                                                                  print(f ""
                                                                                                                          "<!DOCTYPE html>
                                                                                                                        <html lang="en"><head><meta charset="UTF-8"><title> Hello from CGI</title><style> body{{margin: 0;
height : 100vh;
background : #000000;
color : #ffffff;
font - family : Helvetica
    , Arial
    , sans - serif;
display : flex;
align - items : center;
justify - content : center;
}
}
.container {
    { text - align : center; }
}
h1{{font - size : 6rem;
margin : 0;
}
}
.time {
    {
        margin - top : 1rem;
        font - size : 1.5rem;
    opacity:
        0.9;
    }
}
a {
    {
    color:
#ffffff;
        text - decoration : none;
        font - size : 1rem;
    }
}
a : hover{{text - decoration : underline;
}
}
</style>
</head>
<body>
    <div class="container">
        <h1>Hello</h1>
        <div class="time">{current_time}</div>
        <p><a href='/'>Back</a></p>
    </div>
</body>
</html>""")